### ISSUE TITLE
css duplicates

### ISSUE THREAD

**Issue by SummerWish** [kriasoft/isomorphic-style-loader#18]
By using this module, one piece of CSS code exists both in HTML and in JavaScript. Besides, all `require` mentioned CSS stays in JavaScript. The module itself create duplicates (and make HTML much bigger).

**Comment by koistya** [comment]
That's by design. The critical path CSS is loaded first along with the HTML document so that a web page could be rendered even before the full JavaScript/CSS bundle is loaded. Most of the time, this piece of CSS is much smaller than the full version (for all pages/screens) that embedded into JavaScript application bundle.

https://developers.google.com/web/fundamentals/performance/critical-rendering-path/?hl=en

**Comment by SummerWish** [comment]
Thanks. So in fact this module can't reduce size, but can solve flick problems and keep stylesheet inside HTML small?

BTW, how about initialy set `display:none` and switch to `display: block` after components are mounted, which doesn't require initial CSS inside HTML?

**Comment by koistya** [comment]
> Can I understand as that this module can't reduce size, but can solve flick problems and keep stylesheet inside HTML small?

Yep, that's correct. It's supposed to improve the initial page load time and reduce memory footprint occupied by CSS in DOM.

> BTW, how about initialy set display:none and switch to display: block after components are mounted..

This may defeat the purpose of server-side rendering (see [isomorphic apps](https://www.lullabot.com/articles/what-is-an-isomorphic-application)).

**Comment by alex-shamshurin** [comment]
Dublicating styles makes developing and css debugging aanoying. If styles were inserted on server side in the head of the page, why component inserts this again if page already has it? All my styles are repeated twice, is that normal?

**Comment by koistya** [comment]
@alex-shamshurin at this moment, CSS generated on the server can be manually removed on the client once the client app is fully loaded/rendered, for example:

 ```
ReactDOM.render(component, container, () => {
  if (initialRendering) {
    const node = document.getElementById('css');
    node.parent.removeChild(node);
  }
}); 
```
Of cause, it would be nice to handle this automatically, but I'm not sure yet how to implement it correctly. Feel free to share your ideas how this could be accomplished.

**Comment by alex-shamshurin** [comment]
Hmm, I'm not quite sure where to appy your example... As I see in my code page the server rendered styles are inserted in the first <style> tag under <head> and does not have id.  This seems to be a global styles. All subsequent <style> have id's.  
![styles-dubs](https://cloud.githubusercontent.com/assets/1595209/20228702/9b2e8da0-a863-11e6-8a5a-7e79df030dca.jpg)
And at first glance I do noe see repeated styles. But they are:
![css duplicates2](https://cloud.githubusercontent.com/assets/1595209/20228786/08283f8c-a864-11e6-814b-7a3bbcbc9e51.jpg)

**Comment by Everettss** [comment]
@alex-shamshurin what stops you with changing `<style>` to `<style id="css">` on the server side?

**Comment by alex-shamshurin** [comment]
Hmm, `client.js: Uncaught TypeError: Cannot read property 'removeChild' of undefined(…)`

**Comment by alex-shamshurin** [comment]
thank you

**Comment by dmitriykharchenko** [comment]
From the code of `WithStyles` it looks like it inserts styles of a component on mount, and it means that if a component mounted 100 times on a page, css code for that component will get 100 times multiplied.
Any Ideas how to avoid that?

**Comment by alex-shamshurin** [comment]
I move styles of repeated components to parent component (container) in order to include it just once. it will be nice if the library take care of this automaticlally
